src/main/java/liquibase/ext/otbo/changes/DropFunctionGenerator.java

```diff
@@ -0,0 +1,26 @@
+package liquibase.ext.otbo.changes;
+
+import liquibase.database.Database;
+import liquibase.database.core.MSSQLDatabase;
+import liquibase.database.core.OracleDatabase;
+import liquibase.exception.ValidationErrors;
+import liquibase.sql.Sql;
+import liquibase.sql.UnparsedSql;
+import liquibase.sqlgenerator.SqlGeneratorChain;
+import liquibase.sqlgenerator.core.AbstractSqlGenerator;
+
+public class DropFunctionGenerator extends AbstractSqlGenerator<DropFunctionStatement> {
+	public boolean supports( DropFunctionStatement statement, Database database ) {
+		return database instanceof OracleDatabase || database instanceof MSSQLDatabase;
+	}
+	
+	public ValidationErrors validate( DropFunctionStatement statement, Database database, SqlGeneratorChain sqlGeneratorChain ) {
+		ValidationErrors validationErrors = new ValidationErrors();
+		validationErrors.checkRequiredField( "functionName", statement.getFunctionName() );
+		return validationErrors;
+	}
+	
+	public Sql[] generateSql( DropFunctionStatement statement, Database database, SqlGeneratorChain sqlGeneratorChain ) {
+		return new Sql[] { new UnparsedSql( "DROP FUNCTION " + statement.getFunctionName() ) };

```

Comment:
In Oracle I think this is fine, but in MSSQL I wonder if this should be more precise by including the schema name. I was also wondering about whether MSSQL should be putting the names in square brackets to avoid keyword conflicts. i.e. [database_name].[schema_name].[function_name]

Reply:
I'll add an optional schemaName attribute to be used for MSSQL if supplied. Brackets are a good idea; I'll also add them to the function calls for consistency.